examples/ai-core/src/stream-text/openai-fullstream-with-tools.ts

```diff
@@ -0,0 +1,80 @@
+import { experimental_streamText } from 'ai';
+import { openai } from '@ai-sdk/openai';
+import dotenv from 'dotenv';
+import { z } from 'zod';
+import { weatherTool } from '../tools/weather-tool';
+
+dotenv.config();
+
+async function main() {
+  const result = await experimental_streamText({
+    model: openai('gpt-3.5-turbo'),
+    tools: {
+      weather: weatherTool,
+      cityAttractions: {
+        parameters: z.object({ city: z.string() }),
+      },
+    },
+    prompt: 'What is the weather in San Francisco?',
+  });
+
+  for await (const part of result.fullStream) {
+    switch (part.type) {
+      case 'text-delta': {
+        console.log('Text delta:', part.textDelta);
+        break;
+      }
+
+      case 'tool-call': {
+        switch (part.toolName) {
+          case 'cityAttractions': {
+            console.log('TOOL CALL cityAttractions');
+            console.log(`city: ${part.args.city}`); // string
+            break;
+          }
+
+          case 'weather': {
+            console.log('TOOL CALL weather');
+            console.log(`location: ${part.args.location}`); // string
+            break;
+          }
+        }
+
+        break;
+      }
+
+      case 'tool-result': {
+        switch (part.toolName) {
+          // NOT AVAILABLE (NO EXECUTE METHOD)
+          // case 'cityAttractions': {
+          //   console.log('TOOL RESULT cityAttractions');
+          //   console.log(`city: ${part.args.city}`); // string
+          //   console.log(`result: ${part.result}`);
+          //   break;
+          // }
+
+          case 'weather': {
+            console.log('TOOL RESULT weather');
+            console.log(`location: ${part.args.location}`); // string
+            console.log(`temperature: ${part.result.temperature}`); // number
+            break;
+          }
+        }
+
+        break;
+      }
+
+      case 'finish': {
+        console.log('Finish reason:', part.finishReason);
+        console.log('Usage:', part.usage);
+        break;
+      }
+
+      case 'error':
+        console.error('Error:', part.error);
+        break;
+    }
+  }
+}
+
+main().catch(console.error);
```

Comment:
why does this test need to be copied? vs just having openai-fullstream?

Reply:
the logprobs examples could be separate, 2 are enough imo:

- `openai-fullstream-logprobs` (streamText)
- `openai-logprobs` (generateText)

Reply:
I renamed `openai-fullstream` to `openai-fullstream-with-tools` since this uses tools. I then created a new `openai-fullstream` which doesn't call tools, but does console.log the logprobs. Since logprobs and tools are incompatible, I thought it was worth separating into separate examples.

Reply:
agree on the separation. imo logprobs is the less common use case (compared to tools), therefore i'd prefer to separate out logsprobs to a couple of logprobs examples.

Reply:
Sure, will do!